src/main/scala/com/fulcrumgenomics/umi/ConsensusCaller.scala

```diff
@@ -0,0 +1,419 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) $year Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.umi.ConsensusCallerOptions._
+import com.fulcrumgenomics.util.LogDouble._
+import com.fulcrumgenomics.util.LogProbability._
+import com.fulcrumgenomics.util.PhredScore._
+import com.fulcrumgenomics.util.{LogDouble, PhredScore, ProgressLogger}
+import dagr.commons.CommonsDef._
+import htsjdk.samtools._
+import htsjdk.samtools.util.SequenceUtil
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+object ConsensusCallerOptions {
+  /** Various default values for the consensus caller. */
+  val DefaultTag: String                             = "RX"
```

Comment:
`MI` to match `GroupReadsByUmi` unless we want to standardize on something else.